File: c++/2108.cpp
// Link: https://leetcode.com/problems/find-first-palindromic-string-in-the-array/description/
// Runtime: 42 ms
// Memory: 23.24 MB

class Solution 
{
public:
    std::string firstPalindrome(const std::vector<std::string>& words) 
    {
        for (const auto& word : words)
        {
            int i = 0;
            bool isPalindrome = true;
            while ((i < static_cast<int>(word.size() / 2)) && (isPalindrome))
            {
                if (word[i] != word[word.size() - 1 - i])
                    isPalindrome = false;

                i++;
            }

            if (isPalindrome)
                return word;
        }

        return "";
    }
};

File: python/2108.py
# Link: https://leetcode.com/problems/find-first-palindromic-string-in-the-array/description/
# Runtime: 72 ms
# Memory: 16.85 MB


class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            if word == word[::-1]:
                return word

        return ""
